### ISSUE TITLE
Test with new pyarrow dtypes and parquet improvements

### ISSUE THREAD

**Issue by mrocklin** [coiled/coiled-runtime#651]
We made some changes around `nullable_dtypes` and pyarrow strings.  I'd like to see how these affect benchmarks.

cc @jrbourbeau

**Comment by ncclementi** [comment]
Regarding pyarrow strings, you can find some interesting experiments here combined with the use of p2p https://github.com/coiled/h2o-benchmarks/issues/14#issuecomment-1297581487 with the 50GB dataset for the h2o benchmarks. 

The queries references and code is here https://github.com/coiled/h2o-benchmarks/blob/main/h2o_benchmarks_2022-10-0_pyarrow_str_50GB.ipynb

Performance reports https://github.com/coiled/h2o-benchmarks/tree/main/performance-reports-pyarr_str-50GB

**Comment by mrocklin** [comment]
Good to know.  My understanding is that new features like the `use_nullable_dtypes=` flag and the pyarrow filesystem option were merged later.  I'm curious about workflows that never touch Python object dtypes, even from the start of the read_parquet file.  

For context, folks like @crusaderky are looking into how we interact with larger VMs and my hypothesis is that we'll become much less sensitive once we never touch python object dtypes.  This hypothesis is somewhat motivated by [this video](https://www.youtube.com/watch?v=EGgKeZ7_LxQ&ab_channel=Coiled).

**Comment by ncclementi** [comment]
> Good to know. My understanding is that new features like the `use_nullable_dtypes=` flag and the pyarrow filesystem option were merged later. 

Yes, you are right. I think at the moment, we were just casting the types to use `pyarrow_strings` and we saw that memory usage reduce significantly. 

>I'm curious about workflows that never touch Python object dtypes, even from the start of the read_parquet file.
> For context, folks like @crusaderky are looking into how we interact with larger VMs and my hypothesis is that we'll become much less sensitive once we never touch python object dtypes. This hypothesis is somewhat motivated by [this video](https://www.youtube.com/watch?v=EGgKeZ7_LxQ&ab_channel=Coiled).

Thanks for this context, that makes a lot of sense.
